### PR TITLE
Move Groundswell invalid emails to Radbear Multitenant

### PR DESCRIPTION
**Pull request by lukepfeiffer**
Hey,
     I moved the dropped emails functionality from Groundswell to Radbear Multitenant. It works and the test suite passes BUT in order to use RadbearMailer.simple_message  I had to pass in a company, this company being Company.first. I know this is not ideal but was unsure as to what I should have done.

### CODE REVIEW

File: app/controllers/invalid_emails_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    emails.each do |email|
+      puts "#{email} is an invalid email!\n"
+    end
+
+    email = emails.first
+    super_admins = User.super_admins
+
+    subject = "Invalid Email"
+    message = "Someone tried to send an email to #{email} and the email was not properly sent."
+    company = Company.first
```
Comment (plicjo) [inline app/controllers/invalid_emails_controller.rb]: Let's pass in the super company here, which is "Groundswell" in Groundswell (lol), by using the `Company.super_company`.

Great job on being proactive about this issue.

File: app/controllers/invalid_emails_controller.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    from = "no-reply@groundswell.online"
+    recipient = super_admins.map(&:email).first
+
+    RadbearMailer.simple_message(company, recipient, subject, message, from: from).deliver_later
```
Comment (plicjo) [inline app/controllers/invalid_emails_controller.rb]: We want to send this email to all the super admins. Let's loop over them and send emails to all of them. Let's even add another super admin in the test to reflect this requirement.

**Comment by plicjo** [comment]
Unless you expose the invalid_emails routes inside of `config/routes`, this controller is never accessible in Groundswell and will not work.

Add the same route that you added in the dummy app here https://github.com/radicalbear/radbear_multi_tenant/blob/master/config/routes.rb#L4

Please open up a PR in Groundswell that points to this branch in RBMT, deploying a review app that we can test against. We need to be certain that this code will work in GS after we merge.

File: spec/dummy/db/migrate/20170612132336_add_super_company_to_companies.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddSuperCompanyToCompanies < ActiveRecord::Migration[5.0]
+  def change
+    add_column :companies, :super_company, :boolean, default: false
```
Comment (garyfoster) [inline spec/dummy/db/migrate/20170612132336_add_super_company_to_companies.rb]: all boolean columns need a null: false

File: app/controllers/invalid_emails_controller.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
+class EmailError < StandardError; end
+
+class InvalidEmailsController < ApplicationController
+  authority_actions error: "read"
```
Comment (garyfoster) [inline app/controllers/invalid_emails_controller.rb]: is this needed, since this controller is being ignored in auth in application_controller?

**Comment by plicjo** [comment]
There's a lotta BS in this commit history. Duplicated commits, unrelated changes.

@lukepfeiffer We should pair on this branch so that I can show you how to....

# Rewrite History

File: spec/dummy/config/routes.rb [hunk 5]
```diff
@@ -26,4 +26,6 @@
     get :edit, on: :collection
     put :update, on: :collection
   end
+
+  post "/error", to: "invalid_emails#error"
```
Comment (garyfoster) [inline spec/dummy/config/routes.rb]: is this needed? i assumed if we do the routes like we do system messages that each app wouldn't have to configure a route

Comment (plicjo) [inline spec/dummy/config/routes.rb]: @lukepfeiffer Can you make it applicable to all app's like Gary suggests above? Also, can you make this route have a less generic name like "/email_error"?

**Comment by lukepfeiffer** [comment]
I replicated how the `SystemMessagesController` is set up for `ErrorsController` but these features were untested for SystemMessages. The only way I could get a test to pass was to a route to the `spec/dummy/config/routes`. Not sure how to move forward or if this is all right.

**Pull request merged**